oeps/oep-0008.rst

```diff
@@ -0,0 +1,399 @@
+| clean             | delete the artifacts created when running the validations    |
+|                   | and/or the application itself                                |
++-------------------+--------------------------------------------------------------+
+| pip-compile       | update the ``requirements/*.txt`` files with the latest      |
+|                   | packages satisfying ``requirements/*.in``                    |
++-------------------+--------------------------------------------------------------+
+| requirements      | install development environment requirements                 |
++-------------------+--------------------------------------------------------------+
+| requirements.prod | install production requirements                              |
+| (optional)        |                                                              |
++-------------------+--------------------------------------------------------------+
+| requirements.test | install requirements for testing                             |
+| (optional)        |                                                              |
++-------------------+--------------------------------------------------------------+
+| quality           | check coding style                                           |
++-------------------+--------------------------------------------------------------+
+| test              | run all unit, integration, and acceptance tests for the repo |
++-------------------+--------------------------------------------------------------+
+| validate          | run tests and quality checks                                 |
++-------------------+--------------------------------------------------------------+
+| e2e (optional)    | run End to End tests against a deployed environment to       |
+|                   | verify the system as a user, initiating their actions        |
+|                   | through the UI                                               |
++-------------------+--------------------------------------------------------------+
+
+.. _Travis CI: https://travis-ci.org/
+.. _PyPI: https://pypi.python.org/pypi
+.. _Paver: https://pythonhosted.org/Paver/
+.. _Gulp: http://gulpjs.com/
+
+
+Python Language Specific
+------------------------
+
+Requirements
+^^^^^^^^^^^^
+
+* Core requirements are listed in the ``install_requires`` section of
+  ``setup.py``, and only modified when code changes are made that require a
+  new package or drop support for a previously supported one.
+
+* Requirements are managed via `pip-tools`_ and organized by which ones need
+  to be installed for different Python virtual environments.
+
+* Requirements files are kept either at root level or within a directory named
+  ``requirements``.
+
+* Direct dependencies are listed in ``requirements/*.in``.
+
+* Full pinned dependencies are auto-generated via ``make pip-compile`` and
+  saved in ``requirements/*.txt``. This file is tracked under version control.
```

Comment:
You should explicitly talk about the distinction between applications and libraries. In particular:

- libraries shouldn't pin their dependencies in setup.py
- applications should pin all of their dependencies in a requirements.txt file/folder
- libraries should provide a dev-requirements file with the requirements needed to build/develop on the library. That file may or may not be pinned.